### PR TITLE
Hide output with no_log for sensitive variables

### LINKED ISSUE #567: Exposition of sensitive values in ansible outputs
**Describe the issue**
During execution of the playbook there are parts like [this](https://github.com/confluentinc/cp-ansible/blob/0c925803404396d41face98fb881f38140d01747/roles/confluent.kafka_broker/tasks/main.yml#L210) where a secret is printed on output. A best practice for Ansible is to hide sensitive outputs with [`no_log: True`](https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-keep-secret-data-in-my-playbook)

**Expected behaviour**
Secrets and password must not be printed in output to protect them.

**Environment (please complete the following information):**
Any

### PR DESCRIPTION
**Pull request by gianluca-mascolo**
# Description

This fix avoid printing variables on standard output when using shell tasks.
A new variable `mask_secrets` (boolean) is introduced, and tasks with sensitive output are configured with `no_log: "{{mask_secrets}}"`.

Sample output with `mask_secrets: false` (passwords are from scenario `scram-rhel`)
```
TASK [confluent.kafka_broker : Create SCRAM Users] *****************************
changed: [kafka-broker1] => (item={'key': 'admin', 'value': {'principal': 'admin', 'password': 'admin-secret'}})
changed: [kafka-broker1] => (item={'key': 'client', 'value': {'principal': 'client', 'password': 'client-secret'}})
changed: [kafka-broker1] => (item={'key': 'kafka_connect', 'value': {'principal': 'kafka_connect', 'password': 'kafka_connect-secret'}})
changed: [kafka-broker1] => (item={'key': 'control_center', 'value': {'principal': 'control_center', 'password': 'control_center-secret'}})
changed: [kafka-broker1] => (item={'key': 'kafka_rest', 'value': {'principal': 'kafka_rest', 'password': 'kafka_rest-secret'}})
changed: [kafka-broker1] => (item={'key': 'ksql', 'value': {'principal': 'ksql', 'password': 'ksql-secret'}})
changed: [kafka-broker1] => (item={'key': 'schema_registry', 'value': {'principal': 'schema_registry', 'password': 'schema_registry-secret'}})
changed: [kafka-broker1] => (item={'key': 'kafka_connect_replicator', 'value': {'principal': 'kafka_connect_replicator', 'password': 'kafka_connect_replicator-secret'}})
```
Sample output with `mask_secrets: true`
```
TASK [confluent.kafka_broker : Create SCRAM Users] *****************************
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1] => (item=None)
changed: [kafka-broker1]
```
Fixes #567

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

Tested with default scenario `rbac-scram-custom-rhel`

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 1]
```diff
@@ -231,6 +231,7 @@
   loop: "{{ sasl_scram_users|dict2items }}"
   run_once: true
   when: "'SCRAM-SHA-512' in kafka_broker_sasl_enabled_mechanisms"
+  no_log: true
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml:234]: Can you put in the description what the log looks like now? I'm curious the before/after

File: roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore.yml [hunk 2]
```diff
@@ -15,20 +16,23 @@
...
   register: keytool_type
+  no_log: true
 
 - name: Export Certificate from Keystore - PKCS12
   shell: |
     openssl pkcs12 -in {{keystore_path}} \
       -nokeys -out {{cert_path}} \
       -passin pass:{{keystore_storepass}}
   when: keytool_type.stdout == 'PKCS12'
+  no_log: true
```
Comment (domenicbove) [inline roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore.yml:27]: Also wdyt about a variable?
```suggestion
  no_log: "{{mask_passwords_in_logs}}"
```

Or something like that, open to suggestions on the variable name.

Reason being when i debug ssl errors i like copying the shell command onto the host and running it.

File: roles/confluent.kafka_broker/tasks/main.yml (hunk 1 above)
Comment (gianluca-mascolo) [reply roles/confluent.kafka_broker/tasks/main.yml]: done!

File: roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore.yml (hunk 2 above)
Comment (gianluca-mascolo) [reply roles/confluent.ssl/tasks/export_certs_from_keystore_and_truststore.yml]: nice idea, I used a variable named `mask_secrets` (default: `true`).

**Comment by gianluca-mascolo** [comment]
I removed masking from ldap and kerberos tests. It seems to break the tests, i suppose it is because ldapadd cli expect to have password on command line or something similar. This won't affect production feature of the main playbook, secrets will be masked when configuring the cluster.

**Comment by domenicbove** [comment]
Yep, the ldap/kerberos roles are just for testing, those are fine to leave out. This looks great, nice addition!

**Review by domenicbove — APPROVED** [review]
LGTM

**Pull request merged**